test/signhash.ts

```diff
@@ -142,4 +142,66 @@ contract('SignHash', accounts => {
       });
     });
   });
+
+  describe('#removeProof', () => {
+    it('should remove proof method', async () => {
+      const signer = accounts[0];
+      const method = 'http';
+      const value = 'example.com';
+      await instance.addProof(method, value, { from: signer });
+
+      await instance.removeProof(method, { from: signer });
+      const proof = await instance.getProof(signer, method);
+      assert.equal(proof, '');
+    });
+
+    it('should emit ProofRemoved event', async () => {
+      const signer = accounts[0];
+      const method = 'http';
+      const value = 'example.com';
+      await instance.addProof(method, value, { from: signer });
+      const trans = await instance.removeProof(method, { from: signer });
+      const log = findLastLog(trans, 'ProofRemoved');
+      const event: ProofRemoved = log.args;
+
+      assert.equal(event.signer, signer);
+      assert.equal(event.method, method);
+    });
+
+    it('should throw when method is empty', async () => {
+      const signer = accounts[0];
+      const method = '';
+
+      await assertThrowsInvalidOpcode(async () => {
+        await instance.removeProof(method, { from: signer });
+      });
+    });
+
+    it('should throw when proof does not exist', async () => {
+      const signer = accounts[0];
+      const method = 'doesNotExist';
+
+      await assertThrowsInvalidOpcode(async () => {
+        await instance.removeProof(method, { from: signer });
+      });
+    });
```

Comment:
There should also be a test for removing a proof of another signer.

Reply:
Not possible. The proof is always related to `msg.sender`. It would be the same test as `should throw when proof does not exist`.

Reply:
Ok, I can add a test or two

Reply:
Fixed in 094e9ec